src/backend/src/services/teams.services.ts

```diff
@@ -73,9 +73,16 @@ export default class TeamsService {
 
     if (team.dateArchived) throw new HttpException(400, 'Cannot edit the members of an archived team');
 
-    if (team.leads.map((lead) => lead.userId).some((leadId) => userIds.includes(leadId)))
-      throw new HttpException(400, 'team leads cannot be members!');
-
+    // if the new members array includes a current lead on that team, that member will be deleted as a lead of that team
+    const newLeadsArr: number[] = [];
+    if (team.leads.map((lead) => lead.userId).some((leadId) => userIds.includes(leadId))) {
+      for (let i = 0; i < team.leads.length; i++) {
```

Comment:
I like the other for loops being filters, could you do it for this one as well?

Reply:
> I like the other for loops being filters, could you do it for this one as well?

Yeah I think I must've missed this one. I'll fix that now as well.